internal/planner/multi.go

```diff
@@ -136,12 +136,22 @@ func (p *parallelNode) nextMerge(_ int, plan planNode) (bool, error) {
 		return false, err
 	}
 
-	doc := plan.Value()
-	copy(p.currentValue.Fields, doc.Fields)
+	p.currentValue = p.mergeDoc(p.currentValue, plan.Value().Fields)
 
 	return true, nil
 }
 
+func (p *parallelNode) mergeDoc(doc core.Doc, newFields core.DocFields) core.Doc {
+	for i := range newFields {
+		if doc.Fields[i] == nil {
+			doc.Fields[i] = newFields[i]
+		} else if newSubDoc, ok := newFields[i].(core.Doc); ok {
+			doc.Fields[i] = p.mergeDoc(doc.Fields[i].(core.Doc), newSubDoc.Fields)
```

Comment:
todo: Please document why this is needed, it is not terribly clear why the node should be mutating child relations.

Reply:
well, it now justifies the name of method `nextMerge`.

I'll add a comment.

Reply:
I'm not sure the new comment answers what I'm most uncertain about - why does this function recursively mutate child documents, instead of just the top doc?  Why and under which circumstances is the mutation of the children required?  Do please message/call me on discord if you unsure as to what I'm asking.

Reply:
there is no real reason for recursion here. I just though it would cover some future perverted edge-cases.
I made it merge just top fields to keep it simple.

Reply:
Ah got it, thanks Islam!